Makefile

```diff
@@ -27,32 +27,126 @@ escape_double_quote = $(subst $\",$\\",$(1))
 # Utility function to require a variable to be defined
 require_defined = $(if $(value $1),$(value $1),$(error "$1 needs to be defined"))
 
+# What implementation to build
+VARIANT ?= trivial
 # Where to put build files
 OUT ?= out
 # Where to put typescript files
 TS_OUT ?= typescript
 
 .PHONY: all clean test typescript
 
-all: \
-	$(OUT)/baseDAO.tz
+trivialDAO: VARIANT = trivial
+trivialDAO: baseDAO
+
+registryDAO: VARIANT = registry
+registryDAO: baseDAO
+
+treasuryDAO: VARIANT = treasury
+treasuryDAO: baseDAO
+
+testDAO: VARIANT = test
+testDAO: baseDAO
+
+all: $(OUT)/trivialDAO.tz $(OUT)/registryDAO.tz $(OUT)/treasuryDAO.tz $(OUT)/testDAO.tz
 
 # Compile LIGO contract into its michelson representation.
-$(OUT)/baseDAO.tz: src/**
+baseDAO: src/**
```

Comment:
related to the previous comment as well: I see that there is now this target (used by ones above) but that its content is duplicated in the ones below, which seems to be doing the same thing :thinking: 

This was beforehand a single `$(OUT)/baseDAO.tz` target, couldn't these options all be re-united again?
Maybe in a single `$(OUT)/$(VARIANT)DAO.tz` target?

Reply:
Ah, yes.

Reply:
I just could't make it work. 

It appears that the wildcard does not match in more than one way, so when doing `make all`, make throws an error after building the variant for the first match.

Reply:
Ah, I see now, my suggestion isn't actually possible to use.

I looked at the docs and the patterns here aren't actually wildcards, they are variables.
However we can combine them with wildcards to get what we want.

See the comments that I left in the "review" below

Reply:
Worked. Comments were very helpful. Thanks!